sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/UpgradesMatrix.scala

```diff
@@ -1981,12 +2003,14 @@ class UpgradesMatrixCases(val langVersion: LanguageVersion) {
       ).slowAppend(testCase.additionalv2KeyArgsValue(templateDefsV2PkgId, setupData)),
     )
 
-    def globalContractKeyWithMaintainers(setupData: SetupData): GlobalKeyWithMaintainers =
-      GlobalKeyWithMaintainers.assertBuild(
-        v1TplId,
-        globalContractv1Key(setupData),
-        Set(setupData.alice),
-        templateDefsPkgName,
+    def globalContractKeyWithMaintainers(setupData: SetupData): Option[GlobalKeyWithMaintainers] =
+      Option.when(langVersion >= LanguageVersion.Features.contractKeys)(
```

Comment:
Assuming you've applied my other comment:
```suggestion
      ifKeysO(
```